hr_attendance_report_theoretical_time/models/hr_leave.py

```diff
@@ -1,12 +1,48 @@
 # Copyright 2017-2019 Tecnativa - Pedro M. Baeza
+# Copyright 2021 Tecnativa - Víctor Martínez
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from odoo import models
+from odoo import models, fields
 
 
 class HrLeave(models.Model):
     _inherit = 'hr.leave'
 
+    def _prepare_resource_leave_values(self):
+        return {
+            'date_from': fields.Datetime.to_string(
+                fields.Datetime.from_string(self.date_from)
+            ),
+            'date_to': fields.Datetime.to_string(
+                fields.Datetime.from_string(self.date_to)
+            ),
+        }
+
+    def recompute_leave_dates(self):
+        """Recompute leaves if resource_calendar update after create record, although
+        it is strange, it has been possible to change the work schedule"""
+        for item in self:
+            vals = item.read()[0]
+            for key in vals:
+                if isinstance(vals[key], tuple):
+                    vals[key] = vals[key][0]
+            virtual = self.env["hr.leave"].new(vals)
+            virtual._onchange_request_parameters()
+            if (
+                virtual.date_from != item.date_from
+                or virtual.date_to != item.date_to
+            ):
+                item._onchange_request_parameters()
+                # change resource_calendar_leaves
+                resource = self.env['resource.calendar.leaves'].search([
+                    ('holiday_id', '=', item.id)
+                ])
+                if resource:
+                    resource.write(item._prepare_resource_leave_values())
+                # update meeting_id
+                if item.meeting_id:
```

Comment:
Don't do this. This is probably in the past, so this doesn't matter and it has nothing to do with theoretical report.